### PR TITLE
Implement toggle to control istio x-forwarded-for header

### LINKED ISSUE #8855: Having the ability of setting AppendXForwardedHost to false with IstioIntegration enabled
### Gloo Edge Product

Enterprise

### Gloo Edge Version

1.15

### Is your feature request related to a problem? Please describe.

We have a client which have several applications considering the X-Fowarded-Host header.

Once they enabled istio Integration, the calls to this app started failing because it seems that, when you enable istio integration with GE, AppendXForwardedHost is enforced to true: https://github.com/solo-io/gloo/blob/114838123221fb71bd27af1d26f878dc483ecf4b/projects/gloo/pkg/plugins/istio_integration/plugin.go#L54

I would like to know why is this setting enforced to true and if we could modify this behaviour

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by SantoDE**
# Description

Fixes #8855 

With our reworked istio-integration that we introduced with 1.15, its not always necessary to rewrite the xff header. We have some users that run into trouble because of the header being present. Now there is a toggle to control, whether or not users want to have the header.

## API changes
- Added a new field called appendIstioXff to Settings

## Code changes
- Honour the new field and toggle it on the generated envoy config

# Context
Users ran into this, that now in some situations the header was added where it previously might not have been. To get more control over that header, there is now the toggle.

## Interesting decisions
 
We chose to do things this way because ...

## Testing steps

I manually verified behavior by toggling the behaviour through helm installations with a values file like this
```
global:
  istioIntegration:
    enableIstioSidecarOnGateway: true
    disableAutoinjection: true
    appendIstioXff: true
  istioSDS:
    enabled: true
  glooMtls:
    istioProxy:
      image:
        registry: docker.io/istio
        repository: proxyv2
        tag: 1.19.5
    sds:
      image:
        repository: sds
        tag: 1.15.10
```

## Notes for reviewers

Be sure to verify intended behavior by ...

Please proofread comments on ...

This is a complex PR and may require a huddle to discuss ...

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/8855

### CODE REVIEW

File: changelog/v1.17.0-beta3/istio-xff-toggle.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+changelog:
+  - type: NEW_FEATURE
+    issueLink: https://github.com/solo-io/gloo/issues/8855
+    resolvesIssue: true
+    description: >-
+      There are certain situations, where re-writing the X-Forwarded-For header in combination with the istio integration
```
Comment (jenshu) [inline changelog/v1.17.0-beta3/istio-xff-toggle.yaml:6]: typo? the modified code deals with X-Forwarded-Host but in the description and the proto fields it's talking about X-Forwarded-For

File: install/helm/gloo/generate/values.go [hunk 2]
```diff
@@ -755,4 +755,5 @@ type IstioIntegration struct {
...
 	EnableIstioSidecarOnGateway *bool   `json:"enableIstioSidecarOnGateway,omitempty" desc:"Enable Istio sidecar injection on the gateway-proxy deployment. Ignored if LabelInstallNamespace is not 'true'. Ignored if disableAutoinjection is 'true'."`
 	IstioSidecarRevTag          *string `json:"istioSidecarRevTag,omitempty" desc:"Value of revision tag for Istio sidecar injection on the gateway-proxy and discovery deployments (when enabled with LabelInstallNamespace, WhitelistDiscovery or EnableIstioSidecarOnGateway). If set, applies the label 'istio.io/rev:<rev>' instead of 'sidecar.istio.io/inject' or 'istio-injection:enabled'. Ignored if disableAutoinjection is 'true'."`
+	AppendIstioXff              *bool   `json:"appendIstioXff,omitempty" desc:"Controls whether or not to append the XFF header with the istio value. Defaults: false."`
```
Comment (jenshu) [inline install/helm/gloo/generate/values.go:758]:
```suggestion
	AppendIstioXfh              *bool   `json:"appendIstioXfh,omitempty" desc:"Controls whether or not to append the X-Forwarded-Host header with the istio value. Defaults: false."`
```

File: projects/gloo/api/v1/settings.proto [hunk 3]
```diff
@@ -714,6 +714,12 @@ message GlooOptions {
     // Set escapeCharacters for all TransformationTemplates on all vhosts and routes.
     // This setting can be overridden in individual TransformationTemplates
     google.protobuf.BoolValue transformation_escape_characters = 17;
+
+    // Previously, we added the xff header in case of the istio integration by default
+    // However, since there is the enable mtls command now this is not needed anymore
+    // However, there might be cases where users rely on that.
+    // Defaults to false
+    google.protobuf.BoolValue append_istio_xff = 18;
```
Comment (jenshu) [inline projects/gloo/api/v1/settings.proto:722]: xfh

File: install/helm/gloo/templates/18-settings.yaml [hunk 4]
```diff
@@ -20,6 +20,9 @@ spec:
 {{- toYaml .Values.settings.secretOptions.sources | nindent 6 }}
 {{- end }}
   gloo:
+{{- if .Values.global.istioIntegration.appendIstioXff }}
+    appendIstioXff: true
+{{- end }}
```
Comment (jenshu) [inline install/helm/gloo/templates/18-settings.yaml:25]: xfh

File: projects/gloo/api/v1/settings.proto (hunk 3 above)
Comment (jenshu) [inline projects/gloo/api/v1/settings.proto:722]: do we also want to allow overriding at the listener/vhost/route level?

Comment (SantoDE) [reply projects/gloo/api/v1/settings.proto]: Thats a valid question. From a client I only know that they ask for it generally. Hence, I think its a good start with that and eventually make it configurable later?

File: install/helm/gloo/generate/values.go [hunk 5]
```diff
@@ -755,4 +755,5 @@ type IstioIntegration struct {
...
 	EnableIstioSidecarOnGateway *bool   `json:"enableIstioSidecarOnGateway,omitempty" desc:"Enable Istio sidecar injection on the gateway-proxy deployment. Ignored if LabelInstallNamespace is not 'true'. Ignored if disableAutoinjection is 'true'."`
 	IstioSidecarRevTag          *string `json:"istioSidecarRevTag,omitempty" desc:"Value of revision tag for Istio sidecar injection on the gateway-proxy and discovery deployments (when enabled with LabelInstallNamespace, WhitelistDiscovery or EnableIstioSidecarOnGateway). If set, applies the label 'istio.io/rev:<rev>' instead of 'sidecar.istio.io/inject' or 'istio-injection:enabled'. Ignored if disableAutoinjection is 'true'."`
+	AppendIstioXfh              *bool   `json:"appendIstioXfh,omitempty" desc:"Controls whether or not to append the XFF header with the istio value. Defaults: true."`
```
Comment (jbohanon) [inline install/helm/gloo/generate/values.go:758]: nit: 
```suggestion
	AppendIstioXfh              *bool   `json:"appendIstioXfh,omitempty" desc:"Enable appending the X-Forwarded-Host header with the Istio-provided value. Default: true."`
```

File: projects/gloo/api/v1/settings.proto [hunk 6]
```diff
@@ -715,6 +715,12 @@ message GlooOptions {
     // Set escapeCharacters for all TransformationTemplates on all vhosts and routes.
     // This setting can be overridden in individual TransformationTemplates
     google.protobuf.BoolValue transformation_escape_characters = 17;
+
+    // Previously, we added the xff header in case of the istio integration by default
+    // However, since there is the enable mtls command now this is not needed anymore
+    // However, there might be cases where users rely on that.
```
Comment (jbohanon) [inline projects/gloo/api/v1/settings.proto:721]:
```suggestion
    // Set to false to disable adding X-Forwarded-Host header in Istio integration.
```

**Comment by bewebi** [comment]
Closing in favor of #9125

**Pull request closed without merging**